File: Novels/packages/5-mobile-ios/StoryReader/Model/ParagraphComment.swift
import Foundation

// Paragraph Comment model - Duanping (Micro Level)
struct ParagraphComment: Codable, Identifiable {
    let id: String
    let chapterId: String
    let paragraphIndex: Int
    let content: String
    let author: CommentAuthor
    let reactionType: ReactionType?
    let likes: Int
    let createdAt: Date
    let isAuthor: Bool // Story author interaction
    let replies: [ParagraphComment]?
    
    enum CodingKeys: String, CodingKey {
        case id
        case chapterId = "chapter_id"
        case paragraphIndex = "paragraph_index"
        case content, author
        case reactionType = "reaction_type"
        case likes
        case createdAt = "created_at"
        case isAuthor = "is_author"
        case replies
    }
}

struct CommentAuthor: Codable {
    let id: String
    let username: String
    let avatarUrl: String?
    
    enum CodingKeys: String, CodingKey {
        case id, username
        case avatarUrl = "avatar_url"
    }
}

enum ReactionType: String, Codable {
    case like
    case laugh
    case cry
    case angry
    case wow
    case love
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/ParagraphComment.kt
package com.storysphere.storyreader.model

enum class ParagraphReactionType(val emoji: String) {
    LIKE("👍"),
    LAUGH("😂"),
    CRY("😭"),
    ANGRY("😡"),
    WOW("😮"),
    LOVE("❤️")
}

data class ParagraphComment(
    val id: String,
    val chapterId: String,
    val paragraphIndex: Int,
    val userId: String,
    val username: String,
    val avatarUrl: String? = null,
    val content: String,
    val reaction: ParagraphReactionType? = null,
    val reactionBreakdown: Map<ParagraphReactionType, Int> = emptyMap(),
    val likeCount: Int = 0,
    val isAuthor: Boolean = false,
    val createdAt: Long = System.currentTimeMillis()
)


